### PR TITLE
Allow overriding Topiary's configuration

### PR DESCRIPTION
**Pull request by ErinvanderVeen**
This PR makes it a bit easier to configure topiary. It adds four new weeks to provide a configuration file to Topiary.

- `--configuration-override` is a command line argument that overrides ALL configuration of Topiary with the provided file
- `TOPIARY_CONFIGURATION_OVERRIDE=` is an environment variable that overrides ALL configuration of Topiary with the provided file
- `--configuration-file` is a command line argument that adds a file to the list of configuration files that Topiary merges (it has the highest priority)
- `TOPIARY_CONFIGURATION_FILE` is an environment variable that adds a file to the list of configuration files that Topiary merges (it has the second highest priority)

TODO:
- [x] `--configuration-file`
- [x] `TOPIARY_CONFIGURATION_FILE`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ targets = ["x86_64-unknown-linux-gnu", "x86_64-apple-darwin", "x86_64-pc-windows
 [workspace.dependencies]
 assert_cmd = "2.0"
 cfg-if = "1.0.0"
-clap = "4.3"
+clap = { version = "4.3", features = [ "env" ] }
```
Comment (torhovland) [inline Cargo.toml:40]: Interesting that it supports env vars.

Comment (ErinvanderVeen) [reply Cargo.toml:40]: And very nice that we don't have to!

File: topiary-cli/src/configuration.rs [hunk 2]
```diff
@@ -4,27 +4,57 @@ use topiary::{default_configuration_toml, Configuration};
 
 use crate::error::{CLIResult, TopiaryError};
 
-pub fn parse_configuration() -> CLIResult<Configuration> {
-    user_configuration_toml()?
+pub fn parse_configuration(
+    c_override: Option<PathBuf>,
```
Comment (torhovland) [inline topiary-cli/src/configuration.rs:8]: Spelling out `config` (or `configuration`) would be better.

File: topiary-cli/src/configuration.rs [hunk 3]
```diff
@@ -4,27 +4,57 @@ use topiary::{default_configuration_toml, Configuration};
...
-        .fold(default_configuration_toml(), |a, b| {
-            merge_toml_values(a, b, 3)
-        });
+    })
+    .filter_map(|file| {
+        std::fs::read_to_string(file)
+            .map(|config| toml::from_str(&config))
+            .ok()
+    })
+    .collect::<Result<Vec<_>, _>>()?
```
Comment (torhovland) [inline topiary-cli/src/configuration.rs:53]: `collect` followed by `into_iter` looks redundant. Can it be removed?

**Review by torhovland — APPROVED** [review]

**Pull request merged**